File: TESTS/TEST-SKILL-ANALYSE/DIFFERENTS-FILES-TO-TEST/test-sample.java
/*
 * test-sample.java — Resumen del archivo.
 *
 * Lenguaje: Java
 *
 * Clases:
 *   - ScreenUtils   (línea 23)
 *       - ScreenUtils(int width, int height)   (línea 28)
 *       - area()   (línea 33)
 *       - largest(List<ScreenUtils> screens)   (línea 37)
 *       - validate()   (línea 47)
 *       - toString()   (línea 54)
 *
 * [generado por analyze_file]
 */



// Archivo de prueba para analyze_file
package tests;

import java.util.List;

public class ScreenUtils {

    private final int width;
    private final int height;

    public ScreenUtils(int width, int height) {
        this.width = width;
        this.height = height;
    }

    public int area() {
        return width * height;
    }

    public static ScreenUtils largest(List<ScreenUtils> screens) {
        ScreenUtils best = null;
        for (ScreenUtils s : screens) {
            if (best == null || s.area() > best.area()) {
                best = s;
            }
        }
        return best;
    }

    private void validate() throws IllegalStateException {
        if (width <= 0 || height <= 0) {
            throw new IllegalStateException("bad size");
        }
    }

    @Override
    public String toString() {
        return width + "x" + height;
    }
}

File: TESTS/TEST-SKILL-ANALYSE/DIFFERENTS-FILES-TO-TEST/test-sample.js
/*
 * test-sample.js — Resumen del archivo.
 *
 * Lenguaje: JavaScript
 *
 * Clases:
 *   - Monitor   (línea 26)
 *       - constructor(name, width, height)   (línea 27)
 *       - area()   (línea 33)
 *       - fromScreen(screen)   (línea 37)
 *       - describe(prefix)   (línea 41)
 *
 * Funciones:
 *   - pickLargest(monitors)   (línea 46)
 *   - fitWindow(win, mon)   (línea 50)
 *   - clamp(v, lo, hi)   (línea 55)
 *
 * [generado por analyze_file]
 */




// Archivo de prueba para analyze_file
const MAX = 10;

class Monitor {
  constructor(name, width, height) {
    this.name = name;
    this.width = width;
    this.height = height;
  }

  get area() {
    return this.width * this.height;
  }

  static async fromScreen(screen) {
    return new Monitor(screen.name, 1920, 1080);
  }

  describe(prefix) {
    return `${prefix}: ${this.name}`;
  }
}

export function pickLargest(monitors) {
  return monitors.reduce((a, b) => (a.area > b.area ? a : b));
}

const fitWindow = async (win, mon) => {
  win.moveTo(mon.x, mon.y);
};

const helpers = {
  clamp: function (v, lo, hi) {
    return Math.min(hi, Math.max(lo, v));
  },
};
